### ISSUE TITLE
Help wanted: tmux print control chars after startup

### ISSUE THREAD

**Issue by stark-zhang** [tmux/tmux#2844]
### Issue description

As title said, tmux would print `^[[>0;10;1c;` after startup. 

![image](https://user-images.githubusercontent.com/17326718/130406710-7f48c9d2-a62e-4854-b7ad-09938563233e.png)

it can be reproduced without user-defined `.tmux.conf`, just start tmux and exit, and do these action for several times,  the string will be printed. How to eliminate them?

### Required information

Please provide the following information:

* tmux 3.2a with Z-Shell 5.8
* openSUSE Tumbleweed 20210817 in virtual box, connected from windows by SSH for windows 8.1p1
* `xterm-256color`
* how to upload the logs?

**Comment by nicm** [comment]
Is this printed inside or outside tmux? Please attach the logs to this issue:

<img width="995" alt="Screenshot 2021-08-23 at 08 45 24" src="https://user-images.githubusercontent.com/304243/130410047-d67df6b7-1320-4966-b64d-8275cca908df.png">

**Comment by stark-zhang** [comment]
No, it appeared after tmux startup only. Logs are following, I just exit tmux after it started:

[tmux-server-12191.log](https://github.com/tmux/tmux/files/7030615/tmux-server-12191.log)
[tmux-out-12191.log](https://github.com/tmux/tmux/files/7030619/tmux-out-12191.log)
[tmux-client-12189.log](https://github.com/tmux/tmux/files/7030623/tmux-client-12189.log)

**Comment by nicm** [comment]
How did you attach this log? Did you give it time to upload? It doesn't work:
~~~~
$ ftp https://github.com/tmux/tmux/files/7030615/tmux-server-12191.log
Trying 140.82.121.4...
Requesting https://github.com/tmux/tmux/files/7030615/tmux-server-12191.log
ftp: Error retrieving https://github.com/tmux/tmux/files/7030615/tmux-server-12191.log: 404 Not Found
~~~~

**Comment by stark-zhang** [comment]
I dragged and dropped them in writing box. it might be my network issue. Could you access them this time?

[tmux-client-12189.log](https://github.com/tmux/tmux/files/7035804/tmux-client-12189.log)
[tmux-out-12191.log](https://github.com/tmux/tmux/files/7035808/tmux-out-12191.log)
[tmux-server-12191.log](https://github.com/tmux/tmux/files/7035809/tmux-server-12191.log)

**Comment by nicm** [comment]
If you start and detach tmux fast enough and your terminal is slow enough to respond, then the device query string tmux asked for will end up on the terminal. Is that what you are doing?

**Comment by stark-zhang** [comment]
I cannot get the point. how much the time is fast enough to start and detach tmux?
these chars would appear even if I do other operations instead of exiting it.

**Comment by nicm** [comment]
If you do this outside tmux, what do you see:
~~~~
printf '\033[>c\033[>q'

**Comment by stark-zhang** [comment]
![image](https://user-images.githubusercontent.com/17326718/130592306-a9901508-0a80-41c6-8f0b-7787f88e5f54.png)

I got this, and this string will be printed when I input `<CR>`.

**Comment by nicm** [comment]
The problem is that tmux is asking for this sequence to be sent once but your terminal is sending it twice. Does this still happen if you use a different terminal?

**Comment by nicm** [comment]
Can you run tmux like this:
~~~~
strace -ttfostrace.out tmux -f/dev/null -Ltest new
~~~~
Then reproduce and show me the strace.out file.

**Comment by nicm** [comment]
Actually add -vv as well:
~~~~
strace -vvttfostrace.out tmux -f/dev/null -Ltest new

**Comment by stark-zhang** [comment]
I have tried bash(5.1.8) and sh(5.1), and this problem appeared.
[strace.zip](https://github.com/tmux/tmux/files/7038423/strace.zip)

**Comment by nicm** [comment]
bash and sh are not terminals, they are shells. Try a different terminal like putty.

**Comment by stark-zhang** [comment]
got it, I have tried in windows terminal and MobaXterm, this problem appeared.

**Comment by stark-zhang** [comment]
I have tried in MobaXterm and Windows Terminal, this problem appears normally.

**Comment by nicm** [comment]
Do you still see it if you run `tmux -f/dev/null -Ltest new -- cat` instead of your shell?

**Comment by nicm** [comment]
tmux is definitely getting this twice and I do not see any evidence it is asking for it twice:
~~~~
17740 18:12:19.723468 readv(5,  <unfinished ...>
17740 18:12:19.723853 <... readv resumed>[{iov_base="\33[>0;10;", iov_len=8}], 1) = 8
17740 18:12:19.726362 readv(5,  <unfinished ...>
17740 18:12:19.727418 <... readv resumed>[{iov_base="1c\33[>0;", iov_len=7}], 1) = 7
17740 18:12:19.729552 readv(5,  <unfinished ...>
17740 18:12:19.729717 <... readv resumed>[{iov_base="10;1c", iov_len=5}], 1) = 5
~~~~
So you are saying you see this with multiple different terminals?

**Comment by stark-zhang** [comment]
Did you mean this is a bug from shell? 

`tmux -f/dev/null -Ltest new -- cat` will give output like this screen shot in windows terminal.
![image](https://user-images.githubusercontent.com/17326718/130739082-c0bb0519-02d6-42c5-845b-0ed9afbb9668.png)

I also use windows terminal to connect other server and use tmux for working, such as CentOS 7 with bash, and the problem does not appear.

Is there a workaround to avoid this problem?

**Comment by nicm** [comment]
Well I wondered if the shell was trying to bypass tmux and talk to the tty directly. But what you see with cat is even more strange. `\033[>84;0;0c` is what tmux itself sends in response to `\033[c`. So why is that coming from cat?

What if you do `tmux -f/dev/null -Ltest new -- cat /dev/null` so tmux does not run a shell at all?

Is your CentOS 7 box a much older tmux version?

**Comment by stark-zhang** [comment]
it is tmux 3.1c on CentOS 7.
I just tap the `<CR>`, then the second line was printed. and not output with the newer command.

**Comment by nicm** [comment]
So what if you do `tmux -f/dev/null -Ltest new -- /bin/sh --`? Do you see the problem?

**Comment by stark-zhang** [comment]
it appeared, but `\033[>84;0;0c` did not appear.

**Comment by nicm** [comment]
What if you run this outside tmux:
~~~~
script
printf '\033[>c';cat
~~~~
Do you see one or two responses?

**Comment by stark-zhang** [comment]
it will response and print if I tap the `<CR>` once by once. 
The output file of command `script` is empty.

**Comment by nicm** [comment]
What do you mean once by once?

**Comment by stark-zhang** [comment]
`printf` responded me, printed once, if I tapped `<CR>` or did nothing if I gave no input.

**Comment by nicm** [comment]
If you take the `tmux-out*` file you uploaded, https://github.com/tmux/tmux/files/7030619/tmux-out-12191.log, and do this:
~~~~
cat tmux-out-12191.log; printf '\033[m\n\n\n'; cat
~~~~
Do you see the two escape sequences at the end?

**Comment by nicm** [comment]
For example this is what I see in iTerm2:
<img width="1552" alt="Screenshot 2021-08-25 at 09 49 02" src="https://user-images.githubusercontent.com/304243/130759435-fc50b946-a2e0-4ff8-b22e-2c17ba315d70.png">

**Comment by stark-zhang** [comment]
it shows the out2.txt. I ran `cat out2.txt; printf '\033[m\n\n\n'; cat`, and there was no status line. the terminal was the maximum window size(1366 * 768).

the screenshot was sent repeatly because of my network issue, it could make access to Github be not stable.
`

**Comment by nicm** [comment]
Can you try https://github.com/tmux/tmux/files/7030619/tmux-out-12191.log again and make sure it is the same as the first time? Something in the file is making it respond twice.

**Comment by stark-zhang** [comment]
ran with `cat tmux-out-12191.log; printf '\033[m\n\n\n'; cat`, 
![image](https://user-images.githubusercontent.com/17326718/131308533-a431d7a3-1dfb-4c3d-8fa0-5f11a89ae360.png)

**Comment by nicm** [comment]
OK that is the same. Let me take another look at the file.

**Comment by nicm** [comment]
What about this file? [out3.txt](https://github.com/tmux/tmux/files/7075373/out3.txt)

**Comment by stark-zhang** [comment]
![image](https://user-images.githubusercontent.com/17326718/131326243-0aac1308-a457-4ef1-847b-c2f1f5fccb08.png)
screenshot with out3.txt

**Comment by nicm** [comment]
Do you see this if you use putty as well?

Can you reproduce it with `set -g status off` in `.tmux.conf`? If you can please show me the log files from that.

**Comment by stark-zhang** [comment]
sorry for replying late. 
I have tried in PuTTY, the string was not printed on screen. 
And I configured `.tmux.conf` to disable the status bar, the string was not printed neither.

But I don't want to disable status bar, because I need it to manager windows in tmux, any suggestions?

**Comment by stark-zhang** [comment]
![image](https://user-images.githubusercontent.com/17326718/132103975-aed45a29-12b4-4994-a63d-6adb24cb1878.png)

hmmm... Test with windows terminal.
Status bar disabled seems not resolve this issue completely, but the probability seems decreased.

**Comment by nicm** [comment]
Can you reproduce without status line and with `-vv` and show me the tmux log files? For example do:
~~~~
rm tmux*.log
tmux -Ltest kill-server
tmux -Ltest -vvf/dev/null set -g status off \; new
~~~~

This is almost certainly a terminal bug so if we narrow it down we may be able to work around it but we won't be able to fix it in tmux.

**Comment by stark-zhang** [comment]
[tmux-log.zip](https://github.com/tmux/tmux/files/7110347/tmux-log.zip)

OK, I will push an issue to windows terminal if it's a terminal bug.

**Comment by nicm** [comment]
I cannot download this latest zip file, are you sure you uploaded it correctly?

**Comment by stark-zhang** [comment]
[tmux-log.zip](https://github.com/tmux/tmux/files/7149209/tmux-log.zip)
could you re-open this issue? I am sorry for forgetting it.

**Comment by stark-zhang** [comment]
I have pushed the issue to windows terminal as [#11204](https://github.com/microsoft/terminal/issues/11204)

**Comment by elsaco** [comment]
From tmux manual:

```
     escape-time time
             Set the time in milliseconds for which tmux waits after an escape is input to determine if it is part of
             a function or meta key sequences.  The default is 500 milliseconds.

```
@stark-zhang what happen if you add "set -sg escape-time 10" to your .tmux.conf to increase the time from the default 500ms?

@stark-zhang is this happening when connecting to a remote system or only to the local VM. I can't install Virtualbox due to being mutually exclusive with WSL2 running on my local machine. I could not reproduce when launching tmux on a remote system.

**Comment by nicm** [comment]
I don't know, this seems like it is definitely a terminal bug. This is what tmux is sending up to the point it receives the doubled DA response, [out.txt](https://github.com/tmux/tmux/files/7151995/out.txt). But it contains nothing unusual and we have tried removing the most obvious stuff, it is not consistent. This needs someone to see what the terminal actually thinks it is sending.

**Comment by stark-zhang** [comment]
yes, let's take a look what responded from developers of Windows Terminal.
